### ISSUE TITLE
It's breaking the stream

### ISSUE THREAD

**Issue by Ace2213** [GreenMeteor/humhub-patreon-module#7]
### What steps will reproduce the problem?
Using the model

### What is the expected result?
The Patreon widget appearing without conflicting with anything else

### What do you get instead?
Stream stops working.

### Additional info
The only work around is to load Humhub scripts in the header, but that causes 400 and 403 errors.

| Q                | A
| ---------------- | ---
| HumHub version  | 1.3.15 and 1.4 beta
| PHP version | 7.3.11
| Operating system | CentOS 7

https://community.humhub.com/content/perma?id=201723

**Comment by ArchBlood** [comment]
From looking into this, it would seem that it's on @Patreon's side of things.

**Comment by Ace2213** [comment]
Explains why it started happening suddenly a couple of weeks ago.

**Comment by Ace2213** [comment]
I noticed that removing the script and only loading the link doesn't cause this issue, but then the problem becomes the lack of styling. If we can find a way to incorporate the styling manually without using the script, that could bypass the issue entirely.

![image](https://user-images.githubusercontent.com/12226710/68027358-e5a81980-fc90-11e9-88ad-81b8a2e00436.png)

**Comment by Ace2213** [comment]
Here's what I've come up with:
```
<a href="https://www.patreon.com/bePatron?u=insert" data-patreon-widget-type="become-patron-button" data-patreon-widget-type="become-patron-button">
<div style="text-align:center">
<div class="_2KV-widgets-shared--patreonWidgetWrapper">
<div class="fpvra8-0 cjrEzN">
<a class="sc-htoDjs dvMaNv" color="primary" tabindex="0" type="button"href="https://www.patreon.com/bePatron?u=insert" role="button" target="_blank" rel="noopener noreferrer">
<div class="sc-EHOje jlwTfN">
<span class="sc-bdVaJa dfLyku">
<g>
<circle cx="362.589996" cy="204.589996" data-fill="1" id="Oval" r="204.589996"></circle>
<rect data-fill="2" height="545.799988" id="Rectangle" width="100" x="0" y="0"></rect></g></span>
<div class="sc-gzVnrw htefIp"></div><i class="fa fa-patreon"></i>   Become a patron</div></a></div></div></div></a>

```
Not sure why the fa icon isn't showing up, at least for me. And I'm sure it can be done better overall, but this is a perfectly working Patreon button without using their script.
![image](https://user-images.githubusercontent.com/12226710/68030223-b6e17180-fc97-11e9-9af8-7a1d8283f237.png)

**Comment by ArchBlood** [comment]
Are you using FrontAwsome 5 icon? If not then this will happen.

**Comment by Ace2213** [comment]
It was blocked by CSP. I fixed it.
![image](https://user-images.githubusercontent.com/12226710/68056605-56235a80-fcd2-11e9-803c-7971ca304cd9.png)

**Comment by Ace2213** [comment]
The only thing I'm unable to do is center the button within the panel. But that could be because I'm using [Codebox](https://github.com/GreenMeteor/humhub-codebox-module) to test and not adding the code directly to views yet.

**Comment by ArchBlood** [comment]
The Codebox module you should be able to use both HTML and PHP elements, so I'm not sure what you can't center `<center></center>` or use a CSS class call to do this.

**Comment by Ace2213** [comment]
I added the text-align: center tag inline but it didn't work. I tried manipulating it via Chrome devtools with no luck. I just left it as it is because it's not a big deal.

**Comment by Ace2213** [comment]
Center tags aren't working either. 
![image](https://user-images.githubusercontent.com/12226710/68057245-1cebea00-fcd4-11e9-9c43-271e869864c8.png)

**Comment by ArchBlood** [comment]
That's strange I was able to do it.
![Screenshot_2](https://user-images.githubusercontent.com/35392110/68057274-cd092500-fccb-11e9-92c9-5ff11b76aaf4.png)

**Comment by Ace2213** [comment]
Are you using the Patreon module or a custom HTML snippet?

**Comment by ArchBlood** [comment]
I'm using the Patreon module itself.
I just followed this link.
https://www.patreondevelopers.com/t/unable-to-center-patreon-button-on-wordpress-site/783/2

**Comment by Ace2213** [comment]
Yeah I was able to center the Patreon module when I was using it, but for some reason I can't center the custom HTML code I shared above.

**Comment by ArchBlood** [comment]
Using center elements in both modules worked.
![Screenshot_3](https://user-images.githubusercontent.com/35392110/68057439-44d74f80-fccc-11e9-9c41-a501605fb85f.png)

**Comment by Ace2213** [comment]
Are you using my code?

**Comment by ArchBlood** [comment]
I'm using a slightly trimmed one that I had from before we started using the official widget.
```css
.become-patron-button {
  display: inline-block;
  font-family: sans-serif;
  background: #e6461a;
  color: white;
  padding: .5rem 1rem;
  border-radius: 3px;
  text-decoration: none;
  font-weight: 700;
  user-select: none;
  border: 1px solid #e6461a;
}
.patreon-icon {
  padding-right: 0.25rem;
}
.patreon-widget {
  width: 178px !important;
}
```

**Comment by Ace2213** [comment]
I got my code from what gets generated by the script from the official widget. The button and the logo are created using CSS, but I wasn't able to create the logo and that's why I used fa fa-patreon

**Comment by ArchBlood** [comment]
We'll continue to monitor this issue on our end, but seeing as it's an issue on Patreon's end it's probably easier to get it fixed if it's reported to them directly, but it would seem that their [community forums](https://www.patreoncommunity.com/login#welcome) where most of the help comes from is now private.

**Comment by Ace2213** [comment]
My workaround is fully functional though, without breaking anything.

I'll take a look at the Ko-fi module to see if I can do the same there.

**Comment by Ace2213** [comment]
I figured it out. This directive was forcing its location. Once I disabled it, the button got centered.
![image](https://user-images.githubusercontent.com/12226710/68059815-a7384c00-fcdc-11e9-961f-0cfd42990713.png)

I was also able to do the same for the Ko-fi button. It now also works without using the script and without affecting anything else.
![image](https://user-images.githubusercontent.com/12226710/68059846-cc2cbf00-fcdc-11e9-979f-c4b86ccca9d2.png)

**Comment by ArchBlood** [comment]
Please create a p/r for both of these modules and I'll review them. :smiley:

**Comment by Ace2213** [comment]
I also did it for Paypal, if you want to make a separate module or one module for all of them:
![image](https://user-images.githubusercontent.com/12226710/68063224-19fdf300-fcee-11e9-872e-82268eb1217d.png)

```
<form action="https://www.paypal.com/cgi-bin/webscr" method="post" target="_top">
<input type="hidden" name="cmd" value="_donations" />
<input type="hidden" name="business" value="[user-id]" />
<input type="hidden" name="currency_code" value="USD" />
<div class="btn-container" style="background-color: #fff;width: 150px;border-radius: 20px;margin: auto"><input type="image" src="https://www.paypalobjects.com/webstatic/en_US/i/buttons/PP_logo_h_100x26.png" alt="PayPal" " border="0" name="submit" title="PayPal - The safer, easier way to pay online!" alt="Donate with PayPal button" />
<img alt="" border="0" src="https://www.paypalobjects.com/webstatic/en_US/i/buttons/PP_logo_h_100x26.png" alt="PayPal" " width="1" height="1" />
</div></form>
```

**Comment by ArchBlood** [comment]
After reviewing this issue we've come to the conclusion that using the provided Html is not the best solution to this problem, as the module at the moment uses a PHP call (e.g [`<?= $patreonUrl; ?>`](https://github.com/GreenMeteor/humhub-patreon-module/blob/master/widgets/views/patreonframe.php#L18)) to use the widget URL provided by Patreon to a Patreon account and using the provided Html breaks this functionality and makes it unusable. We plan on _**building a new module for multiple options**_ for the future and then the Patreon & Ko-fi modules will be _**permanently be deprecated**_, we've also noticed that there are some issues that HumHub 1.4.beta-1 has brought in that deals with widgets and how they're displayed if external JavaScripts are required by URL to work, this will have to be reported to the @HumHub team directly as even adding the URL to the _**common.php**_ does not fix it sadly.

**Comment by Ace2213** [comment]
Yeah, the javascript issues are the reason I went for a plain HTML solution. 

The Patreon code is ugly and could really use a clean up. That would probably help debugging.

I recently noticed that about half of it is completely useless. This is what I'm running now and I'm sure it can still be optimized further. I can't tell you how much of this CSS is useful. Probably only 10%.

Also I updated my font-awesome package so the icon now is fab fa-patreon. Humhub however still uses an older version that is included so that would be fa fa-patreon.

```
<div class="_2KV-widgets-shared--patreonWidgetWrapper">
      <div class="fpvra8-0 cjrEzN">
             <a class="sc-htoDjs dvMaNv" color="primary" tabindex="0" type="button"href="https://www.patreon.com/bePatron?u=<?= $patreonUrl; ?>" role="button" target="_blank" rel="noopener noreferrer">
                  <div class="sc-EHOje jlwTfN">
                         <i class="fab fa-patreon"></i>&nbsp;&nbsp;Become a Patron!
                  </div>
             </a>
     </div>
</div>
```

> ._2KV-widgets-shared--patreonWidgetWrapper {
>     color: #052d49;
>     font-family: 'America', 'GT America', 'Lato', 'Helvetica Neue', 'Helvetica', 'Arial', sans-serif;
>     font-size: 16px;
>     -webkit-font-smoothing: antialiased;
>     -moz-osx-font-smoothing: grayscale;
>     text-rendering: optimizeLegibility;
> }
> .dvMaNv {
>     -webkit-box-align: center;
>     align-items: center;
>     backface-visibility: hidden;
>     background-color: rgb(232, 91, 70);
>     box-sizing: border-box;
>     cursor: pointer;
>     display: inline-flex;
>     font-weight: 500;
>     height: unset;
>     -webkit-box-pack: center;
>     justify-content: center;
>     position: relative;
>     pointer-events: unset;
>     text-align: center;
>     text-transform: none;
>     user-select: none;
>     white-space: unset;
>     width: 150px;
>     color: rgb(255, 255, 255) !important;
>     font-size: 13px !important;
>     border-radius: 9999px;
>     border-width: 1px;
>     border-style: solid;
>     border-color: rgb(232, 91, 70);
>     border-image: initial;
>     padding: 0.46875rem 1rem;
>     text-decoration: none;
>     transition: all 300ms cubic-bezier(0.19, 1, 0.22, 1) 0s;
> }.dhBTmG {
>     align-self: center;
>     -webkit-box-align: center;
>     align-items: center;
>     display: inline-flex;
>     filter: none;
>     cursor: unset;
>     vertical-align: unset;
>     height: unset;
>     width: unset;
> }.htefIp {
>     height: 1px;
>     width: 0.5rem;
> }.dfLyku svg {
>     align-self: center;
>     fill: transparent;
>     stroke-width: 1.2px;
> }.dfLyku svg * {
>     vector-effect: non-scaling-stroke;
>     transition: all 300ms cubic-bezier(0.19, 1, 0.22, 1) 0s;
> }
